lib/ansible/modules/cloud/amazon/ec2_tag.py

```diff
@@ -144,6 +144,9 @@ def main():
     tags = module.params.get('tags')
     state = module.params.get('state')
 
+    if module.check_mode and state != "list":
+        module.exit_json(skipped=True, msg="remote module (%s) does not support check mode when using %s state" % (module._name, state))
```

Comment:
There's actually no reason this module couldn't support check_mode for absent or present either. 

If we're going to add check_mode to this module, let's do it properly